frontend/packages/console-app/src/components/user-preferences/language/LanguageDropdown.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { QuickStartContext, QuickStartContextValues } from '@patternfly/quickstarts';
+import { Skeleton, SelectOption, Select, SelectVariant } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { getProcessedResourceBundle } from '../../quick-starts/utils/quick-start-context';
+import { supportedLocales } from './const';
+import { usePreferredLanguage } from './usePreferredLanguage';
+
+const LanguageDropdown: React.FC = () => {
+  // resources and calls to hooks
+  const { i18n, t } = useTranslation();
+  const { setResourceBundle } = React.useContext<QuickStartContextValues>(QuickStartContext);
+  const [preferredLanguage, setPreferredLanguage, preferredLanguageLoaded] = usePreferredLanguage();
+  const [dropdownOpen, setDropdownOpen] = React.useState(false);
+  const loaded: boolean = preferredLanguageLoaded;
+
+  const selectOptions: JSX.Element[] = React.useMemo(
+    () =>
+      Object.keys(supportedLocales).map((lang) => (
+        <SelectOption key={lang} value={lang}>
+          {supportedLocales[lang]}
+        </SelectOption>
+      )),
+    [],
+  );
+
+  const selectedLanguage =
+    preferredLanguage ||
+    // handles languages we support, languages we don't support, and subsets of languages we support (such as en-us, zh-cn, etc.)
+    i18n.languages.find((lang) => supportedLocales[lang]);
+
+  // utils and callbacks
+  const onToggle = (isOpen: boolean) => setDropdownOpen(isOpen);
+  const onSelect = (_, selection: string) => {
+    if (selection !== preferredLanguage) {
+      i18n.changeLanguage(selection || 'en');
```

Comment:
in which case selection is going to be a falsy value?

Reply:
I don't think the dropdown will return a falsy value normally, but this was a safety fallback that was used in the already existing Language Dropdown and I decided to keep it as an extra fallback just in case.
we could remove it.